src/compat/get_minimum_ready_state_for_seeking.ts

```diff
@@ -0,0 +1,24 @@
+import { isSafariDesktop, isSafariMobile } from "./browser_detection";
+
+/**
+ * Return the minimum HTMLMediaElement.readyState value that allow to seek.
+ * Starting from HAVE_METADATA (1), seek will no longer raise an exception.
+ * However, on Safari, the HTMLMediaElement.duration
+ * and HTMLMediaElement.buffered may not be initialized at readyState 1, leading
+ * to cases where they can be equals to `Infinity`.
+ * If so, the range in which it is possible to seek is not yet known.
+ * To solve this, the seek should be done after readyState HAVE_CURRENT_DATA (2),
+ * at that time the previously mentioned attributes are correctly initialized and
+ * the range in which it is possible to seek is correctly known.
+ * @param { string|undefined } seekingTime The wanted time to seek. If `undefined`
```

Comment:
For the JSDoc:

  1. Here it's `number`, not `string`
  2. there's no space in the type annotation
  3. Also, there may be a dash between the variable name and its description, that I prefer to add for readability personally, as you wish
  4. In the param documentation, for `undefined`, I don't think just repeating the current implementation has value in this case, it just repeats what the current code does.

I'm wondering about (4), if the `undefined` seeking time notion leading to a `readyState` of `2` is not too specific for the general compat files.
The source `undefined` is returned by `getInitialTime` when it cannot determine the initial time, but I feel that making compat return `2` here so the RxPlayer logic may check later with `getInitialTime` again is awkward: `compat` has no idea what the initial seek logic is.

To me compat should just say: this things doesn't work on the current device, or this value shoul be used on the current device. It's kind of what the function name (`getMinimumReadyStateForSeeking`) is suggesting but checking for the first `getInitialTime`'s return value in it is kind of out-of-place I feel.

For implementation-related details such as "we will re-call `getInitialTime` later", it should be constrained to the location where `getInitialTime` is called IMO.

Reply:
Yes will update the formating / type error, the formatter does not format JSDoc right ? 

I agree with what you are saying with the concept of initial time does not belong to the compat folder.


Do you think this is better ?
```js
// compat
getMinimumReadyStateForSeeking() {
   return isSafari ? 2 : 1;
}

// initial_seek_and_play
const initiallySeekedTime = typeof startTime === "number" ? startTime : startTime();
const minReadyStateForSeeking = getMinimumReadyStateForSeeking();

```

The difference is that it will always seek at readyState 2 depending or not if the seek time is known or undefined